querydsl-core/src/main/java/com/querydsl/core/alias/Alias.java

```diff
@@ -140,8 +140,16 @@
      * @param <D>
      * @return expression
      */
+    @SuppressWarnings("unchecked")
     public static <D> CollectionPath<D, SimpleExpression<D>> $(Collection<D> arg) {
-        return aliasFactory.getCurrentAndReset();
+        final Object current = aliasFactory.getCurrentAndReset();
+        if (arg instanceof CollectionPath) {
+            return (CollectionPath<D, SimpleExpression<D>>) arg; //NOSONAR
+        } else if (arg instanceof ManagedObject) {
+            return (CollectionPath<D, SimpleExpression<D>>) ((ManagedObject) arg).__mappedPath();
```

Comment:
Prefer `__mappedPath()` over the thread local context, for aliases that are shared between threads (i.e. static variables).